decaf377-fmd/src/detection.rs

```diff
@@ -39,8 +39,8 @@ impl DetectionKey {
         // would be nice to avoid this initialization,
         // does collecting into arrays work yet?
         let mut xs = [Fr::default(); MAX_PRECISION];
-        for i in 0..MAX_PRECISION {
-            xs[i] = hkd::derive_private(
```

Comment:
Avoid bounds-check in loop body by using an iterator over pointers to array elements